_includes/scripts.html

```diff
@@ -225,75 +225,76 @@
     const locationSelect = document.getElementById("filter-location");
     const typeSelect = document.getElementById("filter-type");
     const hostSelect = document.getElementById("filter-host");
+    const filterButton = document.getElementById("events-filter-drawer-button"); // Get the filter button
     const posts = document.querySelectorAll(".grid-item");
     const postList = document.querySelector(".grid");
     const emptyStateDiv = document.querySelector(".events-empty-state");
 
     function checkForVisiblePosts() {
-      // Check if any posts are visible
       const anyVisible = Array.from(posts).some(
         (post) => post.style.display !== "none"
       );
       if (anyVisible) {
-        postList.style.display = ""; // Show the <ol> if there are visible posts
-        emptyStateDiv.style.display = "none"; // Hide the empty state div
+        postList.style.display = "";
+        emptyStateDiv.style.display = "none";
       } else {
-        postList.style.display = "none"; // Hide the <ol> if there are no visible posts
-        emptyStateDiv.style.display = "flex"; // Show the empty state div
+        postList.style.display = "none";
+        emptyStateDiv.style.display = "flex";
       }
     }
 
     function filterPosts() {
       const selectedLocation = locationSelect.value;
       const selectedType = typeSelect.value;
       const selectedHostClass = "filter-host-" + hostSelect.value;
+      let filterActive = false; // Flag to track if any filter is active
```

Comment:
@rohan-kadkol I could use some help with this filters script.
1. Filtering by Location > Virtual is not hooked up yet (and maybe it shouldn't be under location to begin with).
2. The notification marker is not working as expected.  You should see the red dot on the filter dropdown button if any filter is active.
3. We should explore narrowing location down even further.  Perhaps after a continent is selected you can then select a country?
4. Can we save the filter options selected so if the user leaves the page and comes back we keep their preferences?